erpnext/edi/doctype/code_list/code_list.py

```diff
@@ -0,0 +1,125 @@
+# Copyright (c) 2024, Frappe Technologies Pvt. Ltd. and contributors
+# For license information, please see license.txt
+
+from typing import TYPE_CHECKING
+
+import frappe
+from frappe.model.document import Document
+
+if TYPE_CHECKING:
+	from lxml.etree import Element
+
+
+class CodeList(Document):
+	# begin: auto-generated types
+	# This code is auto-generated. Do not modify anything in this block.
+
+	from typing import TYPE_CHECKING
+
+	if TYPE_CHECKING:
+		from frappe.types import DF
+
+		canonical_uri: DF.Data | None
+		default_common_code: DF.Link | None
+		description: DF.SmallText | None
+		publisher: DF.Data | None
+		publisher_id: DF.Data | None
+		title: DF.Data | None
+		url: DF.Data | None
+		version: DF.Data | None
+	# end: auto-generated types
+
+	def on_trash(self):
+		if not frappe.flags.in_bulk_delete:
+			self.__delete_linked_docs()
+
+	def __delete_linked_docs(self):
+		self.db_set("default_common_code", None)
+
+		linked_docs = frappe.get_all(
+			"Common Code",
+			filters={"code_list": self.name},
+			fields=["name"],
+		)
+
+		for doc in linked_docs:
+			frappe.delete_doc("Common Code", doc.name, force=1)
```

Comment:
```suggestion
			frappe.delete_doc("Common Code", doc.name)
```

@blaggacao was this intentional? If there are any existing links to such a force-deleted Common Code, the linking form will break.

Reply:
Hm, this looks like a dev-time QoL feature, in hindsight.